snuba/web/rpc/common/debug_info.py

```diff
@@ -0,0 +1,63 @@
+from typing import List
+
+from sentry_protos.snuba.v1.request_common_pb2 import (
+    QueryInfo,
+    QueryMetadata,
+    QueryStats,
+    ResponseMeta,
+    TimingMarks,
+)
+
+from snuba.utils.metrics.timer import Timer
+from snuba.web import QueryResult
+
+
+def extract_response_meta(
+    request_id: str, debug: bool, query_results: List[QueryResult], timers: List[Timer]
+) -> ResponseMeta:
+    query_info = []
+
+    if debug:
+        for query_result, timer in zip(query_results, timers):
+            stats = query_result.extra.get("stats", {}) or {}
+            profile = query_result.result.get("profile", {}) or {}
```

Comment:
Why is the `or {}` part needed?

Reply:
Mypy infers that the result might be `None`. By adding `or {}`, you're ensuring that the result is always a dictionary. Could you clarify the change you're requesting here? I'm not sure I understand the intent behind your question. Do you have an alternative solution?

Reply:
I thought `query_result.extra.get("stats", {})` should automatically handle the default dictionary case. Was curious to know why the extra `or` is needed.